plugins/emc/emc_vnx_block_lun_perfdata

```diff
@@ -0,0 +1,391 @@
+
+=head2 Installation
+
+ The plugin uses SSH to connect to Control Stations. It's possible to use 'nasadmin' user, but it would be better
+ if you create read-only global user by Unisphere Client. The user should have only Operator role.
+ I created "operator" user but due to the fact that Control Stations already had one internal "operator" user,
+ the new one was called "operator1". So be careful.
+ 
+ On munin-node side choose a user which will be used to connect through SSH. Generally user "munin" is ok. Then,
+ execute "sudo su munin -s /bin/bash", "ssh-keygen" and "ssh-copy-id" to both Control Stations with newly created 
+ user.
+ 
+ Make a link from /usr/share/munin/plugins/emc_vnx_dm_basic_stats to /etc/munin/plugins/emc_vnx_dm_basic_stats_<NAME>,
+ where <NAME> is any arbitrary name of your storage system. The plugin will return <NAME> in its answer 
+ as "host_name" field.
+ Assume your storage system is called "VNX5300".
+ 
+ Make a configuration file at /etc/munin/plugin-conf.d/emc_vnx_block_lun_perfdata_VNX5300
+ 
+ [emc_vnx_block_lun_perfdata_VNX5300]
+ user munin							# SSH Client local user
+ env.username operator1						# Remote user with Operator role
+ env.cs_addr 192.168.1.1 192.168.1.2				# Control Stations addresses
+
+=head1 ERRATA
+
+ It counts Queue Length in not fully correct way. We take parameters totally from both SP's, but after we divide them
+ independently by load of SPA and SPB. Anyway, in most AAA / ALUA cases the formula is correct.
+
+=head1 HISTORY
+
+ 09.11.2016 - First Release
+ 26.12.2016 - Compatibility with Munin coding style
+
+=cut
+
+export LANG=C
+TARGET=$(echo "${0##*/}" | cut -d _ -f 6)
+SPALL="SPA SPB"
+NAVICLI="/nas/sbin/navicli"
+
+ssh_check() {
+        ssh -q $username@$1 "/nasmcd/sbin/getreason | grep -w slot_\`/nasmcd/sbin/t2slot\` | cut -d- -f1"
+}
+
+
+check_conf () {
+	if [ -z "$username" ]; then
+		echo "No username ('username' environment variable)!"
+		return 1
+	fi
+
+	if [ -z "$cs_addr" ]; then
+                echo "No control station addresses ('cs_addr' environment variable)!"
+		return 1
+	fi
+
+	#Choosing Cotrol Station. Code have to be "10"
+	for CS in $cs_addr; do
+		if [[ "10" -eq "$(ssh_check $CS)" ]]; then
+			PRIMARY_CS=$CS
+			break
+		fi
+	done
+
+	if [ -z "$PRIMARY_CS" ]; then
+		echo "No alive primary Control Station from list \"$cs_addr\"";
+		return 1
+	fi
+	return 0
+}
+
+if [ "$1" = "autoconf" ]; then
+	check_conf_ans=$(check_conf)
+        if [ $? -eq 0 ]; then
+                echo "yes"
+        else
+                echo "no ($check_conf_ans)"
+        fi
+        exit 0
+fi
+
+check_conf
+if [[ $? -eq 1 ]]; then
+        exit 1;
+fi
+
+SSH="ssh -q $username@$PRIMARY_CS "
+for PROBESP in $SPALL; do
+	$SSH $NAVICLI -h $PROBESP  > /dev/null 2>&1
+	if [ 0 == "$?" ]; then SP="$PROBESP"; break; fi
+done
+
+if [ -z "$SP" ]; then
+	echo "No active Storage Processor found!";
+	exit 1;
+fi
+NAVICLI="/nas/sbin/navicli -h $SP"
```

Comment:
"SP" as in the official abbreviation means "Storage Processor", a block part of the system. Here we choose alive SP.